### ISSUE TITLE
Investigate and enable CI (Travis) on ethersphere/go-ethereum

### ISSUE THREAD

**Issue by nonsense** [ethersphere/swarm#143]

**Comment by nonsense** [comment]
Having spent a bit of time looking at travis, we:
1. Enabled it on ethersphere/go-ethereum repo.
2. Found out that builds are triggered based on the state of the travis.yml file in the PRs.
3. Found out that we can trigger it with a webhook through Travis API.
4. Found out we can have complicated conditional builds based on env variables, and other Travis features.

Ultimately for now we decided to just keep it enabled on ethersphere/go-ethereum and use the travis.yml from ethereum/go-ethereum repo.

In the future we might want to use the Travis API so that we fire up specific builds, rather then rely on the setup from ethereum/go-ethereum.

cc @gbalint
